testing/generate.js

```diff
@@ -94,39 +135,36 @@ export class RetoolApplication {
     await this.page.click('[data-testid="run-all-tests"]', {timeout: 60000})
   }
 
-  // set singleTest to test name if only testing a single test
-  async assertResults(singleTest) {
+  async assertResults(): Promise<string> {
     const actual = {}
-    const expected = {}
-
     const rawResults = await this.page.getAttribute('[data-testid="all-tests-complete"]', 'data-testResults', {timeout: 60000})
     const results = JSON.parse(rawResults)
 
     if (results['tests']) {
       results['tests'].forEach(function (test) {
         const testName = test['name']
-	
-	if (!(singleTest && singleTest !== testName)) {
-          expected[testName] = true
-          actual[testName] = test['passed']
-        }
+        actual[testName] = test['passed']
       })
     }
 
-    expect(actual).toMatchObject(expected)
+    return JSON.stringify(actual)
   }
 
-  // set singleTest to test name if only testing a single test
-  async test(singleTest) {
+  async test(): Promise<string> {
     await this.openEditor()
     await this.runAllTests()
-    await this.assertResults(singleTest)
+    return await this.assertResults()
   }
 }
 
 test.use({ storageState: 'state.json' })
 
-test.describe('${folderName ? folderName.replace("'", "") + '/' : ''}${testAppName}', () => {\n${individualTests}
+const folderName = '${folderName ? folderName.replace("'", "") : ''}'
+const appName = '${testAppName}'
+const resultsDir = 'tmp'
+const resultsPath = resultsDir + '/' + folderName + '-' + appName + '-test-results.json'
```

Comment:
It's best practice to use the `path.join` to construct paths. I think there's also a bug here. If the folder name is empty the filename will start with a dash.